### PR TITLE
Add mutation for project Create/Update

### PR DESCRIPTION
**Pull request by thenav56**
Addresses Mutation for project
## Changes

- Add has_publicly_viewable_leads attribute in project.

Mention related users here if any.

## This PR doesn't introduce any:

- [ ] temporary files, auto-generated files or secret keys
- [ ] n+1 queries
- [ ] flake8 issues
- [ ] `print`
- [ ] typos
- [ ] unwanted comments

## This PR contains valid:

- [ ] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/lead/admin.py [hunk 1]
```diff
@@ -52,6 +57,7 @@ class LeadAdmin(VersionAdmin):
         'project', 'created_by', 'modified_by', 'attachment', 'assignee',
         'source', 'authors', 'author', 'emm_entities', 'lead_group',
     )
+    readonly_fields=('uuid',)
```
Comment (k9845) [inline apps/lead/admin.py]:
```suggestion
    readonly_fields = ('uuid',)
```

File: apps/lead/public_schema.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    public_lead = graphene.Field(
+        PublicLeadMetaType,
+        uuid=graphene.UUID(required=True),
+    )
+
+    @staticmethod
+    def resolve_public_lead(_, info, **kwargs):
+        def _return(lead, project, has_access):
+            _project = project
+            print('>>', lead, project, project and project.is_private, has_access)
```
Comment (k9845) [inline apps/lead/public_schema.py]:
```suggestion
```

File: apps/project/mutation.py [hunk 3]
```diff
@@ -195,6 +241,40 @@ def get_valid_delete_items(cls, info, delete_ids):
...
+        added_regions = [
+            region
+            for region in Region.objects.filter(id__in=regions_to_add).exclude(
+                id__in=existing_regions.values('id')
+            ).order_by('id')
+            if region.public or region.can_modify(info.context.user)
+        ]
+        deleted_regions = list(existing_regions.filter(id__in=regions_to_remove).order_by('id'))
+        assert len(added_regions) <= len(regions_to_add)
+        assert len(deleted_regions) <= len(regions_to_remove)
```
Comment (k9845) [inline apps/project/mutation.py:270]: Is this required?

Comment (thenav56) [reply apps/project/mutation.py:270]: Yes. making sure this is checked.

**Pull request merged**